### PR TITLE
Fixes #12064 - allow empty values for non-required inputs

### PR DESCRIPTION
**Pull request by iNecas**
Also, make the API error message more descriptive, when some issue is there on
provided inputs.

### CODE REVIEW

File: app/models/concerns/foreman_remote_execution/errors_flattener.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        end
+      end
+      errors.map { |key, messages| self.errors.full_message(key, messages.join(', ')) }
+    end
+
+    private
+
+    def invalid_objects_for_attribute(attribute)
+      if self.respond_to?(attribute)
+        invalid_object = self.send(attribute)
```
Comment (ares) [inline app/models/concerns/foreman_remote_execution/errors_flattener.rb]: I think `public_send` should be enough

File: app/models/template_invocation.rb [hunk 2]
```diff
@@ -16,9 +22,10 @@ def to_action_input
   private
 
   def provides_required_input_values
-    required_input_ids = self.template.template_inputs.where(:required => true).pluck(:id)
-    unless (required_input_ids - self.input_values.map(&:template_input_id)).empty?
-      errors.add(:base, _("Not all required inputs have values."))
+    required_inputs = self.template.template_inputs.where(:required => true)
```
Comment (ares) [inline app/models/template_invocation.rb]: `.includes(:input_values)` might be handy because of line below

Comment (ares) [inline app/models/template_invocation.rb]: dismiss, I misread that

**Comment by ares** [comment]
Tested, works nice, just one nit (`public_send`) and good to go.

**Comment by iNecas** [comment]
[test]

File: app/models/concerns/foreman_remote_execution/errors_flattener.rb (hunk 1 above)
Comment (iNecas) [inline app/models/concerns/foreman_remote_execution/errors_flattener.rb]: Good point, new rubocop would have found that as well https://github.com/bbatsov/rubocop/commit/242a761e0cafc2e4634bd62a3362bed2b70fcec2 (but we need to rely on the one in foreman right now)

**Pull request merged**